samples/minimum_criteria.rst

```diff
@@ -0,0 +1,96 @@
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Minimize the use of platform_allow and architecture filters as it limits the scope of testing
+    to the mentioned platforms and architectures. Reference: `Testcase Docs`_
+  * Make use of integration_platforms to limit the scope when there are timing or
+    resource constraints.
+  * Make the sample as generic as possible. Avoid making a sample platform specific unless it is
+    for particular hardware.
+
+  .. _Testcase Docs: https://docs.zephyrproject.org/latest/develop/test/twister.html?highlight=test%20runner#test-cases.
+
+3. Twister should at the least build every sample.
+++++++++++++++++++++++++++++++++++++++++++++++++++
+  * Every sample must have a YAML file.
+
+    **For example:**
+
+    .. code-block:: console
+
+      tests:
+        sample.kernel.cond_var:
+          integration_platforms:
+            - native_posix
+          tags: kernel condition_variables
+          harness: console
+          harness_config:
+            type: one_line
+            regex:
+              - ".*Waited and joined with 3 threads"
+
+  **Guidelines for Samples:**
+    * Minimize the use of platform_allow and architecture filters.
+    * Do not mark the test as build only unless necessary.
+    * Any test case-specific configuration options are added using extra_args or extra_configs
+      options in the YAML file. The prj.conf file should have the in-common configuration options.
+    * There can be multiple testcases in the sample YAML to verify if the feature works with
+      different configurations. For example, different logging modes are tested using multiple
+      test scenarios in samples/subsys/logging/syst.
+    * Sample output can be validated leveraging the harness_config regex option,
+      wherever applicable.
+
+4. A sample should provide sufficient coverage of a subsystem, feature, or module.
+++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++++
+  **DO's:**
+    * Samples should provide integration testing.
+    * Cover the most common and important use cases of the functionality.
+    * Keep the code simple and easy to read. Example: samples/philosophers.
+
+  **DONT's:**
+    * Samples must not test the negative or edge case behaviors.
+    * Must not be unit tests.
+    * Must not incorporate more than one subsystem or feature.
+
+5. Samples must be documented.
+++++++++++++++++++++++++++++++
+  * Samples must have a README clearly explaining the purpose of the sample and its
+    HW requirements and giving sample output, if applicable.
+  * README.rst needs to be included in the sample folder. Example: samples/subsys/foo/README.rst.
+  * Make sure to incorporate the link to README of frequently used samples in the hierarchy
+    of the "Samples & Demos" section of zephyr docs.
+
+  **README Template:**
+    * Overview, if applicable.
+    * Requirements
+    * Building & Running instructions
```

Comment:
I have the wording "Building & Running" because all the README.rst indicate the same for these instructions. Also, I think combining compiling, flashing & running into just -> build instructions may be confusing as they are different.